packages/apps/src/app.plugins.ts

```diff
@@ -81,6 +91,13 @@ export function inject(this: App, plugin: IPlugin) {
       handler = (event: IPluginActivityEvent) => {
         this.onActivity(plugin as ISender, event);
       };
+    } else if (name === 'custom') {
```

Comment:
what would be really cool is if instead of using a static key like `custom` we allowed any string literal and merged the types into app holistically, so I could do something like `app.event('mcp.task`)` (also to avoid name collisions I thinking namespacing the custom events with the app name would be ideal and save us a ton of headaches down the road.

Reply:
That's exactly what this PR is doing:
app.event('mcp.task') is possible right now. 

The namespace idea is something i was thinking about too.... I wonder if we could do it with template strings. WIll think about it more. Too sleepy now